lib/src/teledart/fetch/webhook.dart

```diff
@@ -21,9 +21,35 @@ import 'dart:convert';
 import '../../telegram/telegram.dart';
 import '../../telegram/model.dart';
 
-class Webhook {
+/// Common superclass for Webhooks.
+///
+/// This library can manage webhooks with a standalone http server from
+/// `dart:io` by using the [Webhook] implementation.
+///
+/// Alternatively, you can implement your own webhook by creating a subclass of
+/// [BaseWebhook] and registering it through [Teledart.setCustomWebhook].
+abstract class BaseWebhook {
```

Comment:
I agree, but there's no `interface` keyword in Dart 2 :) Even though this is an abstract class, users will be able to write `implements BaseWebhook` and use this class as an interface.

Reply:
I see. I did a bit of reading, looks like Dart lang treats classes as interfaces (https://dart.dev/samples#interfaces-and-abstract-classes) 
So with the original implementation, you can actually do the following
``` dart
class MyWebhook implements Webhook {
    // overriding properties and functions....
}
```
Then with the changes we made in `teledart.dart`, you can pass `MyWebhook` into `setupWebhook`

Reply:
The problem is that the original `Webhook` class has many properties that we probably won't need in custom implementations: `url`, `secretPath`, `port`, `serverPort`, `max_connections`, `allowed_updates`, `certificate`, `privateKey` and `uploadCertificate`. Since they're public, `MyWebhook` would have to provide getters and setters for them even though they'll never be used. This is why I introduced `BaseWebhook` that only contains methods that really need to be implemented.

Since the mentioned properties are implementation details, we could also make them private, in which case they wouldn't have to be overridden. That makes it more reasonable to implement `Webhook` directly.

Reply:
I reckon we should take an extra step to create an abstract update fetcher class, check out https://github.com/simolus3/TeleDart/pull/1/